win32/tests/malloc_multi_flex_int/malloc_multi_flex_int.c

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdint.h>
+
+#include "testrunnerswitcher.h"
+
+#include "c_pal/malloc_multi_flex.h"
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(TestClassInitialize)
+{
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+}
+
+static int array1_size = 10;
+static int array2_size = 20;
+static int array3_size = 30;
+
+typedef struct INNER_STURCT_TAG
+{
+    uint32_t inner_int_1;
+    uint64_t inner_int_2;
+}INNER_STRUCT;
+
+typedef struct PARENT_STRUCT_TAG
+{
+    uint64_t int_1;
+    uint32_t* array_1;
+    uint32_t int_2;
+    uint64_t* array_2;
+    INNER_STRUCT* array_3;
+    uint32_t int_3;
+}PARENT_STRUCT;
+
+DEFINE_MALLOC_MULTI_FLEX(PARENT_STRUCT, uint32_t, array_1, uint64_t, array_2, INNER_STRUCT, array_3)
+
+TEST_FUNCTION_INITIALIZE(TestMethodInitialize)
+{
+}
+
+TEST_FUNCTION_CLEANUP(TestMethodCleanup)
+{
+}
+
+static void assign_struct(PARENT_STRUCT* test_struct_handle)
+{
+    test_struct_handle->int_1 = 3;
+    test_struct_handle->int_2 = 6;
+    test_struct_handle->int_3 = 9;
+
+    for (int i = 0; i < array1_size; i++)
+    {
+        test_struct_handle->array_1[i] = i;
```

Comment:
>test_struct_handle->array_1[i] = i; [](http://example.com/codeflow?start=8&length=35)

we should probably set all of the bits to really make sure we didn't have some overlap.

Reply:
Not sure how set all bits.

I tried this -

```
size_t total_size = sizeof(PARENT_STRUCT) + (array1_size * sizeof(uint32_t)) + alignof(uint32_t) - 1 + (array2_size * sizeof(uint64_t)) + alignof(uint64_t) - 1 + (array3_size * sizeof(INNER_STRUCT)) + alignof(INNER_STRUCT) - 1;
memset(test_struct_handle, 1, total_size);
```

But this memset causes the program to crash. Need help here.

Reply:
Fixed